clients/src/main/java/org/apache/kafka/common/telemetry/ClientTelemetry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.telemetry;
+
+import org.apache.kafka.common.metrics.MetricsContext;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Optional;
+
+/**
+ * A {@link MetricsReporter} may implement this interface to indicate support for collecting client
+ * telemetry on the client or server side.
+ */
+@InterfaceStability.Evolving
+public interface ClientTelemetry {
+
+    /**
+     * Implemented by the client {@link MetricsReporter} to provide a {@link ClientTelemetrySender}
+     * instance.
+     * <p>
+     * This instance may be cached by the client.
+     * <p>
+     * This method must always be called after the initial call to
+     * {@link MetricsReporter#contextChange(MetricsContext)} on the {@link MetricsReporter}
+     * implementing this interface.
```

Comment:
Not sure if I understand this comment? Can you elaborate?

Reply:
MetricsReporter has message called `contextChange` which actually should be the first method executed on Reporter even prior to init, but can then be executed again whenever context changes. So the comment above just reminds then please do not call this method to fetch ClientTelemetryReceiver prior proper reporter initialization.

Reply:
Well, but it seems to be an comment for somebody writing broker code, not for the user implementing this interface?

Thus, it seems it should not be part of the public JavaDocs -- user implementing a broker plugin won't care, because they don't make these calls?